### PR TITLE
Only show own repositories on the index page for teachers

### PR DESCRIPTION
**Pull request by jorg-vr**
This pull request limits the repository index page to own repositories for non zeus users.

It is only limited on the index page, and not in the policy, as teachers are allowed to see the list of all repositories on other pages. (Eg as filter dropdown to search for exercises)

I also added an empty state to the repository page as this is now relevant.
image

part of #4362

### CODE REVIEW

**Review by bmesuere — APPROVED** [review]
I slightly prefer having a `_html` empty message with a `<br>` instead of the current solution with a `\n` and changing the css behaviour.

Also, I think technically oefeningenrepository's should be one word. Because you're allowed to add a dash to improve readability, I would make it oefeningen-repository's. But I'll let @niknetniko decide.

**Comment by jorg-vr** [comment]
Ik heb de `\n` vervangen door `_html` en `<br\>`. Maar de white-space css aanpassen blijft nodig voor alle text proper te tonen op kleine schermen. 
image

File: config/locales/views/repositories/en.yml [hunk 1]
```diff
@@ -8,12 +8,14 @@ en:
...
     index:
-      title: All exercise repositories
+      title_zeus: All exercise repositories
+      title: Your exercise repositories
       admins: "Edit administrators"
       courses: "Edit allowed courses"
       edit: "Edit"
       delete: "Delete"
       admin: "You are an administrator of this repository"
+      empty_html: "You don't have any repositories yet.<br/>Click on the plus button to create you first repository or ask a repository admin you know to add you as an administrator."
```
Comment (niknetniko) [inline config/locales/views/repositories/en.yml:18]:
```suggestion
      empty_html: "You don't have any repositories yet.<br/>Click on the plus button to create your first repository or ask a repository admin you know to add you as an administrator."
```

File: config/locales/views/repositories/en.yml [hunk 2]
```diff
@@ -8,12 +8,14 @@ en:
...
     new:
       title: Add an exercise repository
       help_html: "You can find a guide with more information on managing repositories in <a href='https://docs.dodona.be/nl/guides/teachers/new-exercise-repo/#_1-een-git-repository-aanmaken'>our documentation</a>."
```
Comment (niknetniko) [inline config/locales/views/repositories/en.yml:21]: Not relevant for this PR, but for some TODO list: this should probably link to the English guide.

File: config/locales/views/repositories/nl.yml [hunk 3]
```diff
@@ -8,12 +8,14 @@ nl:
...
     index:
-      title: Alle oefeningen repository's
+      title_zeus: Alle oefeningen repository's
+      title: Jouw oefeningen repository's
       admins: "Beheerders aanpassen"
       courses: "Toegelaten cursussen aanpassen"
       edit: "Aanpassen"
       delete: "Verwijderen"
       admin: "Je bent beheerder van deze repository"
+      empty_html: "Je hebt nog geen repository's.<br/>Je kan een repository aanmaken via de plus knop, of je kan aan een repository beheerder vragen om je toe te voegen aan een repository."
```
Comment (niknetniko) [inline config/locales/views/repositories/nl.yml:18]:
```suggestion
      empty_html: "Je hebt nog geen repository's.<br/>Je kan een repository aanmaken via de plus-knop, of je kan aan een repository-beheerder vragen om je toe te voegen aan een repository."
```

File: config/locales/views/repositories/nl.yml [hunk 4]
```diff
@@ -8,12 +8,14 @@ nl:
         toggle-label: "Deze repository uitlichten"
       remote_cant_be_edited_html: De clone URL kan niet aangepast worden nadat de repository werd aangemaakt. Contacteer <a href="mailto:dodona@ugent.be">dodona@ugent.be</a> als het toch aangepast moet worden.
     index:
-      title: Alle oefeningen repository's
+      title_zeus: Alle oefeningen repository's
```
Comment (niknetniko) [inline config/locales/views/repositories/nl.yml:11]: This is not used on the actual page, only as the HTML page title:
image

File: config/locales/views/repositories/nl.yml [hunk 5]
```diff
@@ -8,12 +8,14 @@ nl:
...
+      title: Jouw oefeningen repository's
```
Comment (niknetniko) [inline config/locales/views/repositories/nl.yml:12]:
```suggestion
      title: Jouw oefeningenrepository's
```

**Review by niknetniko — CHANGES_REQUESTED** [review]

**Review by niknetniko — APPROVED** [review]

**Pull request merged**